src/RDom.cpp

```diff
@@ -260,16 +260,19 @@ std::ostream &operator<<(std::ostream &stream, const RVar &v) {
 
 /** Emit an RDom in a human-readable form. */
 std::ostream &operator<<(std::ostream &stream, const RDom &dom) {
+    if (!dom.defined()) {
+        return stream << "RDom()";
+    }
+
     stream << "RDom(\n";
     for (int i = 0; i < dom.dimensions(); i++) {
         stream << "  " << dom[i] << "\n";
     }
     stream << ")";
-    Expr pred = simplify(dom.domain().predicate());
```

Comment:
Why remove the simplify?

Reply:
It's a printing routine - it should print the IR as it is, not transformed in any way.